app/controllers/schedule_controller.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class ScheduleController < ApplicationController
+  def save_scheduled_publishing_datetime
+    Document.transaction do
+      document = Document.with_current_edition.lock.find_by_param(params[:id])
+      @edition = document.current_edition
```

Comment:
Don't think we need to set this as a class instance variable as it's not going into any views.